### ISSUE TITLE
mypy says that I can't assign one of Literal's multiple values to Literal-typed variable

### ISSUE THREAD

**Issue by dimaqq** [python/mypy#16791]
Apologies if I misunderstand something about how `Literal` is supposed to be used.
Here's my MRE:

test.py
```py
from typing import Literal

CHEESE = Literal["cheddar", "havarti"]
CHEDDAR = "cheddar"
HAVARTI = "havarti"

tmp: CHEESE = CHEDDAR
```

test it
```command
> pyright test.py
0 errors, 0 warnings, 0 informations
> mypy test.py
test.py:7: error: Incompatible types in assignment (expression has type "str", variable has type "Literal['cheddar', 'havarti']")  [assignment]
Found 1 error in 1 file (checked 1 source file)
```

**Comment by TeamSpen210** [comment]
The problem is that in order to maintain backward compatibility with pre-literal code, unannotated variables get the type `str`, not a `Literal` type. Otherwise code which assigned different strings to a variable would start producing spurious errors. If you manually annotate with `Literal["cheddar"]` it should work fine. Alternatively, if you annotate it with `Final`, it will be inferred as a literal type since no other value could be assigned.

**Comment by dimaqq** [comment]
In other words, mypy doesn't grok that CHEDDAR is meant to be a constant.
After all, all caps is just a convention, some other code could change the value.
Got it.

**Comment by AlexWaygood** [comment]
> In other words, mypy doesn't grok that CHEDDAR is meant to be a constant. After all, all caps is just a convention, some other code could change the value. Got it.

Yup, mypy doesn't use heuristics like that (though it's true that some other type checkers are more eager in assigning literal types to global assignments, that's not the design decision that mypy's taken).

If you want it to be considered a constant by mypy, using `Final` is probably the way to go (and as @TeamSpen210, an explicit `Literal` annotation will also work).
